src/Repositories/TodoRepository.ts

```diff
@@ -0,0 +1,89 @@
+import { RandomUUIDOptions } from 'crypto';
+import { v4 as uuidv4 } from 'uuid';
+import { todoTasks } from "../Models/tempDataStore"
+
+//definition for each task (todo) object
+export interface Todo {
+    id: string
+    name: string
+    createdAt: Date
+    dueDate: string
+    completedStatus: boolean
+}
+
+//interface to implement repository pattern
+export interface RepositoryPI<T> {
+    add(objectToAdd: T);
+    getAll();
+    getById(objectID: string);
+    getByCompletionStatus(completedStatus: boolean);
+    remove(objectToDelete: string);
+    update(objectToChange: string);
```

Comment:
Shouldnt the parameter here be type `T`?

Reply:
Changed to an id to get the task that will be updated and another of type T to pass the payload through